File: code/CS/Apps/MetricView/MetricView/ChartDirector/chartdir_net/ChartDirector/NetWinCharts/VBNetWinCharts/ParamViewer.vb
Public Class ParamViewer
    Inherits System.Windows.Forms.Form

    ' <summary>
    ' ParamViewer Constructor
    ' </summary>
    Public Sub Display(ByVal sender As Object, ByVal e As ChartDirector.WinHotSpotEventArgs)
        ' Add the name of the ChartViewer control that is being clicked
        listView.Items.Add(New ListViewItem(New String() {"source", sender.Name}))

        ' List out the parameters of the hot spot
        Dim key As DictionaryEntry
        For Each key In e.GetAttrValues()
            listView.Items.Add(New ListViewItem(New String() {key.Key, key.Value}))
        Next

        ' Display the form
        ShowDialog()
    End Sub

    ' <summary>
    ' Handler for the OK button
    ' </summary>
    Private Sub OKPB_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles OKPB.Click
        ' Just close the Form
        Close()
    End Sub
    
End Class

File: code/CS/Apps/MetricView/MetricView/ChartDirector/chartdir_net/ChartDirector/NetWinCharts/CSharpWinCharts/ParamViewer.cs
using System;
using System.Windows.Forms;
using System.Collections;

namespace CSharpChartExplorer
{
	/// <summary>
	/// A simple form to list out all parameters passed to ClickHotSpot event
	/// handler for demo purposes.
	/// </summary>
	public partial class ParamViewer : Form
	{
		/// <summary>
		/// ParamViewer Constructor
		/// </summary>
		public ParamViewer()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();
		}

		/// <summary>
		/// ParamViewer Constructor
		/// </summary>
		public void Display(object sender, ChartDirector.WinHotSpotEventArgs e)
		{
			// Add the name of the ChartViewer control that is being clicked
			listView.Items.Add(new ListViewItem(new string[] {"source", 
				((ChartDirector.WinChartViewer)sender).Name}));

			// List out the parameters of the hot spot
			foreach (DictionaryEntry key in e.GetAttrValues())
			{
				listView.Items.Add(new ListViewItem(
					new string[] {(string)key.Key, (string)key.Value}));
			}

			// Display the form
			ShowDialog();
		}
		
		/// <summary>
		/// Handler for the OK button
		/// </summary>
		private void OKPB_Click(object sender, System.EventArgs e)
		{
			// Just close the Form
			Close();
		}
	}
}
